packages/repository/src/decorators/model.ts

```diff
@@ -41,7 +48,7 @@ export function model(definition?: ModelDefinitionSyntax) {
       modelDef.addProperty(p, propertyMap[p]);
     }
 
-    target.definition = modelDef;
+    (<any>target).definition = modelDef;
```

Comment:
Can we change the type of `target` to `Function & {definition?: ModelDefinition}` so that this cast is not needed?